src/main/kotlin/com/github/biomejs/intellijbiome/lsp/BiomeLspServerSupportProvider.kt

```diff
@@ -39,11 +40,22 @@ import org.eclipse.lsp4j.ConfigurationItem
             projectRootDir
         }
 
+        val roots = BiomeServerService.getInstance(project).getServer(file)?.let { server ->
+            server.descriptor.roots.toMutableList().apply {
+                if (!contains(root)) {
+                    // If the server is already running, add the new root directory to the list and stop the server.
```

Comment:
Out of curiosity, why stopping the server is needed here?

Reply:
oh, I've found a race condition here when two files are opened simultaneously, need to investigate here.

Reply:
Hm, I'm not sure how to change roots without restarting =(